integration-tests/tests/cli/schema.spec.ts

```diff
@@ -355,3 +355,72 @@ describe.each([ProjectType.Stitching, ProjectType.Federation, ProjectType.Single
     );
   },
 );
+
+test.concurrent(
+  'schema:publish with --target parameter matching the access token',
+  async ({ expect }) => {
+    const { createOrg } = await initSeed().createOwner();
+    const { inviteAndJoinMember, createProject, organization } = await createOrg();
+    await inviteAndJoinMember();
+    const { createTargetAccessToken, project, target } = await createProject();
+    const { secret } = await createTargetAccessToken({});
+
+    const targetSlug = [organization.slug, project.slug, target.slug].join('/');
+
+    await expect(
+      schemaPublish([
+        '--registry.accessToken',
+        secret,
+        '--author',
+        'Kamil',
+        '--target',
+        targetSlug,
+        'fixtures/init-schema.graphql',
+      ]),
+    ).resolves.toMatchInlineSnapshot(`
+      :::::::::::::::: CLI SUCCESS OUTPUT :::::::::::::::::
+
+      stdout--------------------------------------------:
+      ✔ Published initial schema.
+      ℹ Available at http://__URL__
+    `);
+  },
+);
+
+test.concurrent(
+  'schema:publish fails with --target parameter not matching the access token',
+  async ({ expect }) => {
+    const { createOrg } = await initSeed().createOwner();
+    const { inviteAndJoinMember, createProject } = await createOrg();
+    await inviteAndJoinMember();
+    const { createTargetAccessToken } = await createProject();
+    const { secret } = await createTargetAccessToken({});
+
+    const targetSlug = 'i/do/not-match';
+
+    await expect(
+      schemaPublish([
+        '--registry.accessToken',
+        secret,
+        '--author',
+        'Kamil',
+        '--target',
+        targetSlug,
+        'fixtures/init-schema.graphql',
+      ]),
+    ).rejects.toMatchInlineSnapshot(`
+      :::::::::::::::: CLI FAILURE OUTPUT :::::::::::::::
+      exitCode------------------------------------------:
+      2
+      stderr--------------------------------------------:
+       ›   Error: No access (reason: "Missing permission for performing 
+       ›   'schemaVersion:publish' on resource")  (Request ID: __REQUEST_ID__)  [115]
+       ›   > See https://__URL__ for
+       ›    a complete list of error codes and recommended fixes.
+       ›   To disable this message set HIVE_NO_ERROR_TIP=1
+       ›   Reference: __ID__
+      stdout--------------------------------------------:
+      __NONE__
+    `);
```

Comment:
@jdolle Is there a difference between `Reference: __ID__` and `__REQUEST_ID__`? Do you see the need to introduce a special error code for permission errors?

Reply:
I don't see a big difference in the id types. I think it was convenient to match on the word prior but they are all IDs.

And if this error is returned from our api then technically no need. But i did envision standardized api errors in the future and i did add unique errors for authorization which obviously is from the api...

Reply:
Oh I understand now, Reference is the error code, Request ID is the actual request id 😓